ccl/changefeedccl: TestAlterChangefeedColumnFamilyDatabaseScope failed

ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope [failed](https://mesolite.cluster.engflow.com/invocations/default/cc396bfe-f79f-41c6-803a-ef265fd580ad?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [6a5fe354bfce1df461ed824d25ff6deadf5a85ac](https://github.com/cockroachdb/cockroach/commits/6a5fe354bfce1df461ed824d25ff6deadf5a85ac):


```
    alter_changefeed_test.go:1225: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterChangefeedColumnFamilyDatabaseScope832273881
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope (3.23s)
=== RUN   TestAlterChangefeedColumnFamilyDatabaseScope/webhook
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.drivers WITH OPTIONS (diff, split_column_families, envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1198: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (2) expected
        	            	  |   drivers.onlyid: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers.onlyname: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	  | got
        	            	  |   drivers: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedColumnFamilyDatabaseScope/webhook
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope/webhook (3.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedColumnFamilyDatabaseScope.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51448

ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope [failed](https://tanzanite.cluster.engflow.com/invocations/default/ee6a9497-36e0-4c70-a4d7-f6d197555beb?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [a0362142bb1903793a7a9327c4c5ac8011aa3dbc](https://github.com/cockroachdb/cockroach/commits/a0362142bb1903793a7a9327c4c5ac8011aa3dbc):


```
    alter_changefeed_test.go:1225: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterChangefeedColumnFamilyDatabaseScope884369841
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope (13.94s)
=== RUN   TestAlterChangefeedColumnFamilyDatabaseScope/webhook
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.drivers WITH OPTIONS (diff, split_column_families, envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1198: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (2) expected
        	            	  |   drivers.onlyid: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers.onlyname: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	  | got
        	            	  |   drivers: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedColumnFamilyDatabaseScope/webhook
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope/webhook (13.94s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedColumnFamilyDatabaseScope.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19930505?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19930505?buildTab=artifacts#/) on master @ [c5954702f1e0a46b20595cbcc7f6a66380f2e6c1](https://github.com/cockroachdb/cockroach/commits/c5954702f1e0a46b20595cbcc7f6a66380f2e6c1):


```
    alter_changefeed_test.go:1225: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8664cbf2d4e437e0e06ce871fe9210c9/logTestAlterChangefeedColumnFamilyDatabaseScope2357032844
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope (1.86s)
=== RUN   TestAlterChangefeedColumnFamilyDatabaseScope/webhook
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.drivers WITH OPTIONS (diff, split_column_families, envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1198: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) expected
        	            	  |   drivers.onlyid: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers.onlyname: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	  | got
        	            	  |   drivers: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedColumnFamilyDatabaseScope/webhook
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope/webhook (1.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedColumnFamilyDatabaseScope.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope [failed](https://tanzanite.cluster.engflow.com/invocations/default/de6ec7dc-4105-47b5-bacf-cdc7226a69f6?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [5bb5e004b5ad8bf7d6436466ba998251fa2f9117](https://github.com/cockroachdb/cockroach/commits/5bb5e004b5ad8bf7d6436466ba998251fa2f9117):


```
    alter_changefeed_test.go:1225: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterChangefeedColumnFamilyDatabaseScope1983802970
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope (4.53s)
=== RUN   TestAlterChangefeedColumnFamilyDatabaseScope/webhook
    helpers_test.go:1238: making server as system tenant
    helpers_test.go:1314: making webhook feed factory
    helpers_test.go:1100: forcing enriched envelope for *changefeedccl.webhookFeedFactory - EXPERIMENTAL CHANGEFEED FOR TABLE movr.drivers WITH OPTIONS (diff, split_column_families, envelope = 'enriched', enriched_properties = 'source')
    alter_changefeed_test.go:1198: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	expected
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBaseErr
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:415
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:440
        	            	  | github.com/cockroachdb/cockroach/pkg/util/timeutil.RunWithTimeout
        	            	  | 	pkg/util/timeutil/timeout.go:28
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.withTimeout
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:436
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloadsBase
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:274
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.assertPayloads
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:447
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.TestAlterChangefeedColumnFamilyDatabaseScope.func1
        	            	  | 	pkg/ccl/changefeedccl/alter_changefeed_test.go:1198
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamed.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1486
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.cdcTestNamedWithSystem.func1
        	            	  | 	pkg/ccl/changefeedccl/helpers_test.go:1521
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (2) expected
        	            	  |   drivers.onlyid: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers.onlyname: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	  | got
        	            	  |   drivers: [1]->{"after": {"id": 1}, "before": null}
        	            	  |   drivers: [1]->{"after": {"name": "Alice"}, "before": null}
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestAlterChangefeedColumnFamilyDatabaseScope/webhook
--- FAIL: TestAlterChangefeedColumnFamilyDatabaseScope/webhook (4.52s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedColumnFamilyDatabaseScope.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/alter_changefeed_test.go
+++ b/pkg/ccl/changefeedccl/alter_changefeed_test.go
@@ -513,8 +513,11 @@ func TestAlterChangefeedDropTargetFamily(t *testing.T) {
 		sqlDB := sqlutils.MakeSQLRunner(s.DB)
 		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING, FAMILY onlya (a), FAMILY onlyb (b))`)
 
-		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo FAMILY onlya, foo FAMILY onlyb`,
-			optOutOfMetamorphicEnrichedEnvelope{"requires families"})
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "metamorphic enriched envelope does not support column families for webhook sinks"})
+		}
+		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo FAMILY onlya, foo FAMILY onlyb`, args...)
 		defer closeFeed(t, testFeed)
 
 		feed, ok := testFeed.(cdctest.EnterpriseTestFeed)
@@ -1192,7 +1195,11 @@ func TestAlterChangefeedColumnFamilyDatabaseScope(t *testing.T) {
 			`INSERT INTO movr.drivers VALUES (1, 'Alice')`,
 		)
 
-		testFeed := feed(t, f, `CREATE CHANGEFEED FOR movr.drivers WITH diff, split_column_families`)
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "metamorphic enriched envelope does not support column families for webhook sinks"})
+		}
+		testFeed := feed(t, f, `CREATE CHANGEFEED FOR movr.drivers WITH diff, split_column_families`, args...)
 		defer closeFeed(t, testFeed)
 
 		assertPayloads(t, testFeed, []string{

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -1913,8 +1913,13 @@ func TestNoBackfillAfterNonTargetColumnDrop(t *testing.T) {
 		sqlDB.Exec(t, `CREATE TABLE hasfams (id int primary key, a string, b string, c string, FAMILY id_a (id, a), FAMILY b_and_c (b, c))`)
 		sqlDB.Exec(t, `INSERT INTO hasfams values (0, 'a', 'b', 'c')`)
 
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "metamorphic enriched envelope does not support column families for webhook sinks"})
+		}
+
 		// Open up the changefeed.
-		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c`)
+		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c`, args...)
 		defer closeFeed(t, cf)
 		assertPayloads(t, cf, []string{
 			`hasfams.b_and_c: [0]->{"after": {"b": "b", "c": "c"}}`,
@@ -1951,8 +1956,12 @@ func TestChangefeedColumnDropsWithFamilyAndNonFamilyTargets(t *testing.T) {
 		sqlDB.Exec(t, `INSERT INTO hasfams values (0, 'a', 'b', 'c')`)
 		sqlDB.Exec(t, `INSERT INTO nofams values (0, 'a', 'b', 'c')`)
 
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "metamorphic enriched envelope does not support column families for webhook sinks"})
+		}
 		// Open up the changefeed.
-		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c, TABLE nofams`)
+		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c, TABLE nofams`, args...)
 		defer closeFeed(t, cf)
 		assertPayloads(t, cf, []string{
 			`hasfams.b_and_c: [0]->{"after": {"b": "b", "c": "c"}}`,
@@ -2000,8 +2009,13 @@ func TestChangefeedColumnDropsOnMultipleFamiliesWithTheSameName(t *testing.T) {
 		sqlDB.Exec(t, `INSERT INTO hasfams values (0, 'a', 'b', 'c')`)
 		sqlDB.Exec(t, `INSERT INTO alsohasfams values (0, 'a', 'b', 'c')`)
 
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "metamorphic enriched envelope does not support column families for webhook sinks"})
+		}
+
 		// Open up the changefeed.
-		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c, TABLE alsohasfams FAMILY id_a`)
+		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c, TABLE alsohasfams FAMILY id_a`, args...)
 		defer closeFeed(t, cf)
 		assertPayloads(t, cf, []string{
 			`hasfams.b_and_c: [0]->{"after": {"b": "b", "c": "c"}}`,
@@ -2089,8 +2103,13 @@ func TestNoStopAfterNonTargetAddColumnWithBackfill(t *testing.T) {
 		sqlDB.Exec(t, `CREATE TABLE hasfams (id INT PRIMARY KEY, a STRING, b STRING, c STRING, FAMILY id_a (id, a), FAMILY b_and_c (b, c))`)
 		sqlDB.Exec(t, `INSERT INTO hasfams values (0, 'a', 'b', 'c')`)
 
+		var args []any
+		if _, ok := f.(*webhookFeedFactory); ok {
+			args = append(args, optOutOfMetamorphicEnrichedEnvelope{reason: "metamorphic enriched envelope does not support column families for webhook sinks"})
+		}
+
 		// Open up the changefeed.
-		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c WITH schema_change_policy='stop'`)
+		cf := feed(t, f, `CREATE CHANGEFEED FOR TABLE hasfams FAMILY b_and_c WITH schema_change_policy='stop'`, args...)
 
 		defer closeFeed(t, cf)
 		assertPayloads(t, cf, []string{
@@ -3323,7 +3342,7 @@ func TestChangefeedEachColumnFamily(t *testing.T) {
 		sqlDB.Exec(t, `DELETE FROM foo WHERE a>0`)
 
 		// Deletes send a message for each column family.
-		fooWithDiff := feed(t, f, `CREATE CHANGEFEED FOR foo WITH split_column_families, diff`)
+		fooWithDiff := feed(t, f, `CREATE CHANGEFEED FOR foo WITH split_column_families, diff`, args...)
 		defer closeFeed(t, fooWithDiff)
 		sqlDB.Exec(t, `DELETE FROM foo WHERE a=0`)
 		assertPayloads(t, fooWithDiff, []string{
@@ -3336,7 +3355,7 @@ func TestChangefeedEachColumnFamily(t *testing.T) {
 		// Table with a second column family added after the changefeed starts.
 		sqlDB.Exec(t, `CREATE TABLE bar (a INT PRIMARY KEY, FAMILY f_a (a))`)
 		sqlDB.Exec(t, `INSERT INTO bar VALUES (0)`)
-		bar := feed(t, f, `CREATE CHANGEFEED FOR bar`)
+		bar := feed(t, f, `CREATE CHANGEFEED FOR bar`, args...)
 		defer closeFeed(t, bar)
 		assertPayloads(t, bar, []string{
 			`bar: [0]->{"after": {"a": 0}}`,

